Stable versions of manifests on 2.3.0

jbech-linaro opened the pull request:
Change-Id: Ifde84bf0e582c76b9a27a0ce64a71e07c30d966e
Signed-off-by: Joakim Bech <joakim.bech@linaro.org>

jbech-linaro commented:
Part of this is done by a script Pascal wrote, but I've manually re-added the `upstream"foo"` everywhere. Strictly I don't think that is necessary to have, but on the other hand it's a nice way to explicitly say where (which branch) the commit comes from.

jforissier commented on dra7xx_stable.xml:
```diff
@@ -0,0 +1,20 @@
+  <project name="hello_world.git" path="hello_world" remote="linaro-swg" revision="a17da737d15438ebedf400255570afb5afe9454f" upstream="master"/>
+  <project name="ti-linux-kernel.git" path="linux" remote="ti-linux" revision="7c580a51af521f320eb56717f291aa5b64c2f244"/>
+  <project name="ti-optee-client.git" path="optee_client" remote="ti-optee" revision="ce5fefe6cba5df9b4f31399b1155ba768eee0b78"/>
+  <project name="ti-optee-os.git" path="optee_os" remote="ti-optee" revision="4556015010dbf648d236a72857e927317b0f6438"/>
```
Why a private Git for `optee_os`? Wouldn't upstream work here?

jbech-linaro replied:
That's a question I've raised with TI and we should try to move toward the official tree's (and send the delta upstream). But for now we have to use their tree instead.

jforissier replied:
So basically we're claiming (upstream) support for DRA7xx since version 1.0.0 and it's not true? :(

IMO it's just wrong to point to anything here other than the Linaro Gits for optee_os, optee_client, and optee_test. Doing otherwise means that our upstream code is probably tested by nobody, which is bad.

jforissier commented on dra7xx_stable.xml:
```diff
@@ -0,0 +1,20 @@
+  <project name="ti-linux-kernel.git" path="linux" remote="ti-linux" revision="7c580a51af521f320eb56717f291aa5b64c2f244"/>
+  <project name="ti-optee-client.git" path="optee_client" remote="ti-optee" revision="ce5fefe6cba5df9b4f31399b1155ba768eee0b78"/>
+  <project name="ti-optee-os.git" path="optee_os" remote="ti-optee" revision="4556015010dbf648d236a72857e927317b0f6438"/>
+  <project name="ti-optee-test.git" path="optee_test" remote="ti-optee" revision="6203b8769beff4eccf7104a6474991114691bf90"/>
```
Same question here: why not use the upstream `optee_test`?

jbech-linaro replied:
Same answer :)

jbech-linaro commented:
QEMU-v7:
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m default_stable.xml
```
Build: OK
Test: OK

jforissier commented:
Looks good although I did not check each and every SHA1. And let's put aside my rant about DRA7xx ;)

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

jbech-linaro commented:
**FVP**
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m fvp_stable.xml
```
Build: OK
Test: OK

**Juno** (Jens)
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m juno_stable.xml
```
Build: OK
Test: OK

**QEMU v8**
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m qemu_v8_stable.xml
```
Build: OK
Test: OK

jbech-linaro commented:
> @jforissier 
>  And let's put aside my rant about DRA7xx ;)

Well, the comment is definitely valid, since it's not a true 2.3.0 release for that particular device. But I hope that we with help from Andrew can switch sooner than later to use the official OP-TEE gits.

jbech-linaro commented:
**DRA7xx**
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m dra7xx_stable.xml
```
Build: OK
Test: OK

jbech-linaro commented:
**HiKey**
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m hikey_stable.xml
```
Build: OK
Test: OK

jbech-linaro commented:
**MTK8173**
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m hikey_stable.xml
```
Build: OK
Test: FAILS! OP-TEE not probed at all. Needs further investigation.

jbech-linaro commented:
@jforissier With patch https://github.com/linaro-swg/linux/pull/31 MTK8173 also works. So we need to merge that patch and then I'll need to update the kernel commit on all xyz_stable.xml files.

jforissier commented:
@jbech-linaro OK

jbech-linaro commented:
**HiKey Debian**

With the review patch applied:
```bash
repo init -u https://github.com/jbech-linaro/manifest.git -b stable_2.3.0 -m hikey_stable.xml
```
Build: OK
Test: OK

jbech-linaro commented:
@jforissier  I think this is enough tested now, after merging this particular commit we should also tag **manifest.git** with 2.3.0.

Squashed, rebased and tag(s) applied (should be) ready for merge! Thanks!

jforissier commented:
Merged and tag created. Thanks!
